api/storage_service.cc

```diff
@@ -444,10 +443,14 @@ void set_repair(http_context& ctx, routes& r, sharded<repair_service>& repair) {
         // returns immediately, not waiting for the repair to finish. The user
         // then has other mechanisms to track the ongoing repair's progress,
         // or stop it.
-        return repair_start(repair, validate_keyspace(ctx, req->param),
-                options_map).then([] (int i) {
-                    return make_ready_future<json::json_return_type>(i);
-                });
+        try {
+            int res = co_await repair_start(repair, validate_keyspace(ctx, req->param), options_map);
+            co_return json::json_return_type(res);
+        } catch (const std::runtime_error& e) {
+            // if the option is not sane, repair_start() throws immediately, so
+            // convert the exception to an HTTP error
```

Comment:
The repair_start could return runtime_error even if the option is supported.  So std::runtime_error does not always mean it fails due to unsupported options.

Reply:
@asias ic. how about changing the exception type of the exception thrown due to unsupported option (or their combination) to `std::invalid_argument`? and catch it instead?

Reply:
Yes. This should work. 

However, even if we return HTTP 500 (Internal Server Error) in case the option is not supported, why would the test code care too much it is not HTTP 400 (Bad Request)?

Reply:
@asias it's more about test. the test case mentioned in the above comment is just an use case of this RESTful API. the reason i mentioned it is to explain why i started looking into it. the reason why i created this PR is to make the semantic of the RESTful API: 500 means internal server error, while 400 implies invalid argument. in the same spirit of following function: https://github.com/scylladb/scylladb/blob/3363f6e1e80fb5e123faa958d006a7dd6b392e90/api/storage_service.cc#L80-L87 , this change intends to expose a better RESTful API with a correct HTTP status code.

Reply:
Also with the new native nodetool, on error, the response from the server is printed to the console. So it is better to be as clear as possible.